src/app/entity/entity.ts

```diff
@@ -23,25 +23,62 @@
  */
 export class Entity {
 
-  private _id: string;
+  /**
+   * The database id in the form <code>prefix:entityId</code>
+   */
+  private readonly _id: string;
 
+  /**
+   * The unique id of this entity.
+   */
+  private readonly entityId: string;
+
+  /**
+   * The prefix for this entity type.
+   */
+  private readonly prefix: string;
+
+  /**
+   * Creates an entity object with the given id. This id is final and won't be changeable after this object has been
+   * created.
+   *
+   * @param id a unique id for this entity
+   */
   constructor(id: string) {
-    this.setId(id);
+    this.entityId = id;
+    this.prefix = this.constructor.name;
+    this._id = this.prefix + ':' + this.entityId;
   }
 
+  /**
+   * Returns the id of this entity.
+   *
+   * Note that an id is final and can't be changed after the object has been instantiated, hence there is no
+   * <code>setId()</code> method.
+   *
+   * @returns {string} the unique id of this entity
+   */
+  public getEntityId(): string {
```

Comment:
Does `Entity` in `getEntityId()` add any useful information? (rather than `getId()`)